### PR TITLE
Address comments for LSTM PR

### PR DESCRIPTION
**Pull request by Vimalakirti**
Addressed comments in the merged PR:
https://github.com/uiuc-kang-lab/zk-torch/pull/44

### CODE REVIEW

File: src/basic_block/constant.rs [hunk 1]
```diff
@@ -41,6 +41,8 @@ impl BasicBlock for Const2BasicBlock {
   }
 }
 
+// ConstOfShapeBasicBlock is a basic block that creates a constant tensor of a given shape and value.
+// It requires no proving since the constant value is known as a public input.
 #[derive(Debug)]
 pub struct ConstOfShapeBasicBlock {
```
Comment (Vimalakirti) [inline src/basic_block/constant.rs:47]: ddkang: I don't know what this is
: added comments for explanation

File: src/basic_block/eq.rs [hunk 2]
```diff
@@ -43,21 +43,22 @@ impl BasicBlock for EqBasicBlock {
   }
 }
 
+// ElementwiseEqBasicBlock is a basic block that performs elementwise equality comparison.
 #[derive(Debug)]
 pub struct ElementwiseEqBasicBlock;
 impl BasicBlock for ElementwiseEqBasicBlock {
   fn run(&self, _model: &ArrayD<Fr>, inputs: &Vec<&ArrayD<Fr>>) -> Vec<ArrayD<Fr>> {
     assert!(inputs.len() == 2 && inputs[0].ndim() <= 1 && inputs[1].ndim() <= 1);
```
Comment (Vimalakirti) [inline src/basic_block/eq.rs:51]: ddkang: Comments
: added comments

File: src/basic_block/range.rs [hunk 3]
```diff
@@ -3,6 +3,9 @@ use ark_bn254::{Fr, G1Affine, G1Projective, G2Affine, G2Projective};
 use ndarray::{arr1, ArrayD, IxDyn};
 use rand::rngs::StdRng;
 
+// RangeBasicBlock is a basic block that creates a tensor of a range of values.
+// It should not be used without proving the range is correct.
+// To prove the range, CQ basic block is used now.
 #[derive(Debug)]
 pub struct RangeBasicBlock {
```
Comment (Vimalakirti) [inline src/basic_block/range.rs:11]: ddkang: Does this need to be proved to be secure?
: Yes, it does and we can prove it by CQ check. Added comments for secure usage.

File: src/layer/constantofshape.rs [hunk 4]
```diff
@@ -6,6 +6,8 @@ use ark_bn254::Fr;
 use ndarray::ArrayD;
 use tract_onnx::pb::AttributeProto;
 
+// Generate a tensor with a given value (the value is in the ONNX attribute) and shape (the shape is in the input tensor)
+// reference: https://onnx.ai/onnx/operators/onnx__ConstantOfShape.html
 pub struct ConstOfShapeLayer;
 impl Layer for ConstOfShapeLayer {
```
Comment (Vimalakirti) [inline src/layer/constantofshape.rs:12]: ddkang: Don't know what this does
: added comments

File: src/basic_block/range.rs (hunk 3 above)
Comment (ddkang) [reply src/basic_block/range.rs:11]: Doesn't the range need to be in a specific order?

File: src/basic_block/eq.rs [hunk 5]
```diff
@@ -43,21 +43,22 @@ impl BasicBlock for EqBasicBlock {
...
     let mut r = ArrayD::zeros(IxDyn(&[std::cmp::max(inputs[0].len(), inputs[1].len())]));
-    if inputs[0].len() == 1 && inputs[1].ndim() > 0 {
+    if inputs[0].len() == 1 && inputs[1].ndim() > 0 { // broadcast inputs[0] to compare with each element in inputs[1]
```
Comment (ddkang) [inline src/basic_block/eq.rs]: Put the comments above, not inline like this

File: src/basic_block/range.rs (hunk 3 above)
Comment (Vimalakirti) [reply src/basic_block/range.rs:11]: You are right, I will address this issue (https://github.com/uiuc-kang-lab/zk-torch/issues/58) in the next PR. Besides, I also added a new comment to note this security issue.

**Review by ddkang — APPROVED** [review]

**Pull request merged**